### ISSUE TITLE
Compatibility with Django1.5?

### ISSUE THREAD

**Issue by mhb11** [Koed00/django-whoshere#2]
@Koed00 
I have a project that uses Django 1.5. Your instructions say whoshere works for 1.7 and above. Are there any changes I can do to make this plugin compatible for 1.5? Please advise.

**Comment by Koed00** [comment]
I've never tested it with 1.5, mainly because it uses the new migrations. Other than that I don't think there are big incompatibilities. So the biggest hurdle would be making the migrations work with South.

**Comment by mhb11** [comment]
@Koed00 fair enough, I'll try it out. BTW to give you more background, my app is a simple chatting web application for people in low-resource areas who have flaky internet and low-end devices. It uses Django 1.5. I wanted to add a feature whereby users can see who else is online at a given time. I suppose your plugin fits the bill? I'm going to look at the migrations issue now.

**Comment by mhb11** [comment]
@Koed00 ah it seems there's more to it than meets the eye. apps.py imports AppConfig from django.apps. This module doesn't exist in Django 1.5. The relevant parts of the traceback are:

File "/home/hassan/.virtualenvs/redditpk/local/lib/python2.7/site-packages/django_whoshere/apps.py", line 1, in <module>
    from django.apps import AppConfig
ImportError: No module named apps

Any suggestions?

**Comment by Koed00** [comment]
I will see if I can fabricate a workaround. Can't promise anything though.

**Comment by mhb11** [comment]
Yea I understand - you're probably swamped with work. But if you can tweak it, would be a tremendous assist! Alternatively, you can guide me vis-a-vis it too.

**Comment by Koed00** [comment]
Well it's more that I don't know what else could come up. Django 1.5 is quite old and considered insecure. I can't even really test for it. The dev branch has a try-except around the AppConfig import. This will disable naming the admin link.  Do you want to give it a try?

**Comment by mhb11** [comment]
Yep for sure I want to give it a shot. Thing is, my interest is in solely using the template tags functionality you've provided. Anything that feeds into that is important, everything else is not necessary.

**Comment by Koed00** [comment]
As far as I can tell the only incompatibilities are with the admin, migrations and the way the apps are loaded.  You really only need the model, middleware and the template tags. Shouldn't be hard to copy them over to your own project. There is no magic.

One thing I would do is install GeoIP instead of Telize for ip locations. Telize gets very laggy if you have more than a few dozen users online. GeoIP doesn't have this problem since it's local.

**Comment by mhb11** [comment]
Roger that. I have started getting a feelsky of things. Just using a select number of features at the moment (i.e. trying to show user ids that are online at the moment, nothing more).

E.g. in models.py, I've added:
from links.models import UserSession
class UserSession(User):

```
def active_user_ids():

    return [user.id for user in UserSession.active_users()]

def active_users():

    return [user for user in UserSession.objects.all() if user.track]

def active_user_count():

    return len(UserSession.active_user_ids())
```

In templatetags/whoshere.py, I've added:
from django import template

#from django_whoshere.apps import get_ip, parse, get_city, get_country
from links.models import UserSession

register = template.Library()

@register.simple_tag(name='active_user_count')
def user_count():
    return UserSession.active_user_count()

@register.assignment_tag(name='active_users')
def active_users():
    return UserSession.active_users()

@register.simple_tag(name='your_agent', takes_context=True)
def user_agent(context):
    agent = context['request'].META.get('HTTP_USER_AGENT', '')
    if parse:
        return parse(agent)
    return agent

and in my base.html, I added:
{% load whoshere %}
{% active_users as users %}
{% for user in users %}
   {{ user.username }} ({{ user.email }})

## {% endfor %}

Note: indentation has been done properly.

I'm getting the error: unbound method active_users() must be called with UserSession instance as first argument (got nothing instead)

Any ideas what to do?

**Comment by Koed00** [comment]
Sounds like the UserSession model is broken, but I can't really tell what you're doing to in in the first import.
